src/mono/wasm/wasm.proj

```diff
@@ -84,12 +84,27 @@
           AfterTargets="Build"
           DependsOnTargets="BuildPInvokeTable;BundleTimeZones">
 
-    <Exec Command="make -C $(MonoProjectRoot)wasm all SHELL=/bin/bash BINDIR=$(ArtifactsBinDir) MONO_BIN_DIR=$(MonoArtifactsPath) OBJDIR=$(ArtifactsObjDir) NATIVE_BIN_DIR=$(NativeBinDir) CONFIG=$(Configuration) PINVOKE_TABLE=$(WasmPInvokeTablePath) ICU_LIBDIR=$(PkgMicrosoft_NETCore_Runtime_ICU_Transport)/runtimes/browser-wasm/native/lib ENABLE_ES6=$(WasmEnableES6)"
+    <PropertyGroup>
+      <_IcuLibdir>$(PkgMicrosoft_NETCore_Runtime_ICU_Transport)/runtimes/browser-wasm/native/lib</_IcuLibdir>
+    </PropertyGroup>
+
+    <Exec Command="make -C $(MonoProjectRoot)wasm all SHELL=/bin/bash BINDIR=$(ArtifactsBinDir) MONO_BIN_DIR=$(MonoArtifactsPath) OBJDIR=$(ArtifactsObjDir) NATIVE_BIN_DIR=$(NativeBinDir) CONFIG=$(Configuration) PINVOKE_TABLE=$(WasmPInvokeTablePath) ICU_LIBDIR=$(_IcuLibdir) ENABLE_ES6=$(WasmEnableES6)"
           IgnoreStandardErrorWarningFormat="true" />
 
     <Copy SourceFiles="$(NativeBinDir)dotnet.js;
                        $(NativeBinDir)dotnet.wasm;
-                       $(NativeBinDir)dotnet.timezones.blat"
+                       $(NativeBinDir)dotnet.timezones.blat;
+                       $(MonoProjectRoot)wasm\runtime\driver.c;
+                       $(MonoProjectRoot)wasm\runtime\pinvoke.c;
+                       $(MonoProjectRoot)wasm\runtime\corebindings.c;
+                       $(MonoProjectRoot)wasm\runtime\pinvoke.h;
+                       $(MonoProjectRoot)wasm\runtime\binding_support.js;
+                       $(MonoProjectRoot)wasm\runtime\dotnet_support.js;
+                       $(MonoProjectRoot)wasm\runtime\library_mono.js;
+                       $(LibrariesProjectRoot)Native\Unix\System.Native\pal_random.js;
+                       $(ArtifactsObjDir)wasm\pinvoke-table.h;
```

Comment:
The header files should not be in the root folder we don't want them in `publish` output. They will be used during before final publish copy step, right?

Reply:
In AOT, only "data" files have to be copied as-is to the output, and the dotnet.wasm and dotnet.js files are not used. The rest will be ingested by emscripten.

Reply:
We need the intermediate products for AOT, they don't strictly need to be as is.

Reply:
Would they be better off in a separate AOT package?

Reply:
We need roughly the same things for unmanaged linking as this PR indicates.